File: 프로그래머스/Level 1/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    
    map<string, int> m;
    
    for (int i = 0; i < participant.size(); i++) {
        m[participant[i]]++;
    }
    
    for (int i = 0; i < completion.size(); i++) {
        m[completion[i]]--;
    }
    
    map<string, int>::iterator it;
    
    for (it = m.begin(); it != m.end(); it++) {
        if (it->second == 1)
            return it->first;
    }
    
    return answer;
}

File: 프로그래머스/Level 1/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        HashMap<String, Integer> map = new HashMap<>();
        
        for (String str : participant) {
            if (map.containsKey(str) == true)
                map.put(str, map.get(str) + 1);
            else
                map.put(str, 1);
        }
        
        for (String str : completion) {
            if (map.get(str) == 1)
                map.remove(str);
            else
                map.put(str, map.get(str) - 1);
        }
        
        for (String str : map.keySet())
			answer = str;
        
        return answer;
    }
}
